src/tests/cli_test.cpp

```diff
@@ -46,15 +46,15 @@ int main(int argc, char** argv) {
   w.add<std::string>("-f,--file", "help message")->check(validators::file_exists);
   //w.add<color>("-e,--enum", "help message for enum");
 
-  w.parse(argc, argv);
+  auto opts = w.parse(argc, argv);
 
-  w.get<int>("int");
-  w.get<std::string>("str");
-  w.get<std::vector<float>>("vec");
-  w.get<std::vector<std::string>>("defvec");
-  w.get<bool>("bool");
+  opts["int"].as<int>();
```

Comment:
What does this do? Check the types by reading them out and ignoring the ret val, which will throw if the type is incorrect? Then maybe add a comment // validate types

Reply:
New way with a proper wrapper. 

```c++
  auto options = New<Options>();
  {
    auto cli = New<cli::CLIWrapper>(options, "Allowed options");
    cli->add<std::string>("--from,-f", "Input model", "model.npz");
    cli->add<std::string>("--to,-t", "Output model", "model.bin");
    cli->parse(argc, argv);
  }
  auto modelFrom = options->get<std::string>("from");
  auto modelTo = options->get<std::string>("to");
```